### DISCUSSION TITLE
Is there a use case (or consequences) for defining the `ractor::Actor` struct with state?

### DISCUSSION THREAD

**Discussion by stuartcarnie** [slawlor/ractor#271] [Q&A] [answered]
All the examples define the actor as an empty structure:

```rust
struct Actor;

impl ractor::Actor for Actor {
...
}
```

Are there consequences to adding immutable state to the `Actor` struct? Would it impact other patterns or processes in ractor?

**Comment by slawlor** [comment, marked as answer]
Nope it's fine to do technically and has no impact but it does mean you need to construct that immutable state outside of the actor spawning flow which does introduce a panic risk to the parent actor. Since it won't be managed for you. 

It's the motivation for why the state is a separate struct, which I think there's a note on the main readme. But for simple things with no risk, it's completely fine as long as you don't need mutability
